tests/xharness/Logging/CaptureLog.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.IO;
+
+namespace Xharness.Logging {
+
+	// A log that captures data written to a separate file between two moments in time
+	// (between StartCapture and StopCapture).
+	public class CaptureLog : Log
+	{
+		public string CapturePath { get; private set; }
+		public string Path { get; private set; }
+
+		long startPosition;
+		long endPosition;
+		bool entire_file;
+		bool started;
+
+		public CaptureLog (ILogs logs, string path, string capture_path, bool entire_file = false)
+			: base (logs)
+		{
+			if (path == null)
+				throw new ArgumentNullException (nameof (path));
+
+			Path = path;
+			CapturePath = capture_path;
+			this.entire_file = entire_file;
+		}
+
+		public void StartCapture ()
+		{
+			if (entire_file)
+				return;
+
+			if (File.Exists (CapturePath))
+				startPosition = new FileInfo (CapturePath).Length;
+			started = true;
+		}
+
+		public void StopCapture ()
+		{
+			if (!started && !entire_file)
+				throw new InvalidOperationException ("StartCapture most be called before StopCature on when the entire file will be captured.");
+			if (!File.Exists (CapturePath)) {
+				File.WriteAllText (Path, $"Could not capture the file '{CapturePath}' because it doesn't exist.");
+				return;
+			}
+
+			if (entire_file) {
+				File.Copy (CapturePath, Path, true);
+				return;
+			}
```

Comment:
I think it should be responsibility of just one method to write the output file like the `Capture`/`Finalize` one.
I suggest for entire_file we just set start to 0 and end to Length and let either `Capture` not even know we have something like entire_file OR `Capture` can do the `Copy`